### PR TITLE
Add `log_path` option to the carthage action

### PR DESCRIPTION
**Pull request by taisukeh**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context

Carthage has the `--log-path` option to specify xcode build log path.

```
% carthage help bootstrap
...
[--log-path (string)]
        path to the xcode build output. A temporary file is used by default
...
```

But the carthage action has no  option to specify it.

### Description

Add `log_path` option to the carthage action.

### CODE REVIEW

File: fastlane/lib/fastlane/actions/carthage.rb [hunk 1]
```diff
@@ -42,6 +43,10 @@ def self.validate(params)
         if command_name != "archive" && params[:output]
           UI.user_error!("Output option is available only for 'archive' command.")
         end
+
+        unless %w(build bootstrap update).include?(command_name)
+          params[:log_path] = nil
```
Comment (joshdholtz) [inline fastlane/lib/fastlane/actions/carthage.rb]: Can we call `UI.user_error!(` instead so users know / aren't surprised why its getting cleared 🙃

**Review by joshdholtz — CHANGES_REQUESTED** [review]
One small thing!

**Comment by taisukeh** [comment]
@joshdholtz Thanks for your feedback. I fixed it.

**Review by joshdholtz — CHANGES_REQUESTED** [review]
One more small nit pick 😉 then we will be 💯

**Comment by taisukeh** [comment]
@joshdholtz I fixed it :bow:

**Review by joshdholtz — APPROVED** [review]
Looks great! Thanks 😊

**Pull request merged**